lib/dotcom/alerts/disruptions/subway.ex

```diff
@@ -46,6 +48,38 @@ defmodule Dotcom.Alerts.Disruptions.Subway do
     |> Enum.into(%{})
   end
 
+  # Given an alert with multiple active periods, some of which may be
+  # contiguous with one another (next one starts the same day or the
+  # day after the previous one), combines the contiguous active
+  # periods, and then returns
+  defp separate_active_periods(alert) do
+    alert.active_period
+    |> combine_contiguous_active_periods()
+    |> Enum.map(fn active_period ->
+      %Alerts.Alert{alert | active_period: [active_period]}
```

Comment:
question: if the original alert has `active_period: []` does this result in `%Alerts.Alert{alert | active_period: [nil]}`? or maybe just `[]`?!

(I actually don't know how possible/probable such an alert is, but I did find other code in Dotcom which presumes it 💀 I'm theorizing it might happen for cancellations. In any case I'd expect an output with an alert with a `[]` active_period rather than a `[nil]` one) 😅

Reply:
I think this would drop the alert, so

```elixir
[%Alert{active_period: []}]
|> split_by_discontiguous_active_periods()
```

would return
```elixir
[]
```

While not intentional, I think that's the right behavior??

(I also suspect that there's code earlier on that will either crash or prevent any alerts with `[]` active_periods from getting to this spot.)

Reply:
OK, I think that makes sense. And if a new scenario of alert proves to be an issue, we can cross that bridge when we get to it